lib/src/builder/generation/declaration_parsing.dart

```diff
@@ -96,33 +117,24 @@ class ParsedDeclarations {
       member.metadata.forEach((annotation) {
         final name = annotation.name.toString();
         if ((isPropsClass(name) || isStateClass(name)) && member is ClassDeclaration) {
-          if (member.name.name.startsWith(generatedPrefix)) {
-            final companionName = member.name.name.substring(generatedPrefix.length);
-            final privateCompanionName = '$privatePrefix$companionName';
-            final privateCompanionClass = unit.declarations.firstWhere(
-                    (innerMember) =>
-                innerMember is ClassDeclaration && innerMember.name.name == privateCompanionName,
-                orElse: () => null);
-            final publicCompanionClass = unit.declarations.firstWhere(
-                    (innerMember) =>
-                innerMember is ClassDeclaration && innerMember.name.name == companionName,
-                orElse: () => null);
-
-            if (privateCompanionClass == null && publicCompanionClass == null) {
-              error('${member.name.name} must have an accompanying companion class within the '
-                  'same file for Dart 2 builder compatibility, but one was not found.', getSpan(sourceFile, member));
-            } else {
-              if (privateCompanionClass != null) {
-                validateMetaField(privateCompanionClass, isPropsClass(name) ? 'PropsMeta' : 'StateMeta');
-              } else {
-                validateMetaField(publicCompanionClass, isPropsClass(name) ? 'PropsMeta' : 'StateMeta');
-              }
-            }
+          final companionName = member.name.name.substring(companionPrefix.length);
+          final companionClass = unit.declarations.firstWhere(
+                  (innerMember) =>
+              innerMember is ClassDeclaration && innerMember.name.name == companionName,
+              orElse: () => null);
+
+          final hasCompanionClass = companionClass != null;
+          if (hasCompanionClass) {
+            updateCompanionClass(name, true);
+            validateMetaField(companionClass, isPropsClass(name) ? 'PropsMeta': 'StateMeta');validateMetaField(companionClass, isPropsClass(name) ? 'PropsMeta': 'StateMeta');
```

Comment:
Looks like this statement was duplicated

Reply:
@evanweible-wf I was thinking about that this weekend. I don't think we should error, since there's really no issue with that class defining a static meta field in the `_$` prefixed class and the extending class. I think a warning is appropriate.